Hide backup scheduled or delayed cards if the backups are stopped

rustamshaikh opened the pull request:
#### Proposed Changes

- This hides the backup scheduled or delayed cards based on the storage usage level.
- If the storage usage level is Full, these cards will be hidden. The usage level defaults to 'Normal' when not set.
- Uses `:empty` CSS selector to hide the card because of the component hierarchy

#### Testing Instructions

- For a site whose backups are stopped due to storage full
  - The Backup Scheduled card should be hidden
   
   Before the changes, the Backup Scheduled were visible even when backups were stopped
   <img width="755" alt="Screenshot 2023-01-25 at 2 49 32 PM" src="https://user-images.githubusercontent.com/13975226/214525762-026fe553-80ba-4fa0-9d0f-cf27e231fa54.png">
   
   After the changes, it should be hidden for such cases
   <img width="747" alt="Screenshot 2023-01-25 at 2 50 38 PM" src="https://user-images.githubusercontent.com/13975226/214525932-0fa49061-3638-474b-ae32-2e097450ebee.png">

   
  - When selecting a previous date other than today but after which backups were stopped, check that `The backup attempt for <selected_date> was delayed' card should be hidden.
  
  Before the changes, when selecting Yesterday as date
  <img width="736" alt="Screenshot 2023-01-25 at 2 49 54 PM" src="https://user-images.githubusercontent.com/13975226/214526417-96fb8fc7-061b-4324-bea8-116e6fd70a0e.png">

  After the changes
  <img width="747" alt="Screenshot 2023-01-25 at 2 50 49 PM" src="https://user-images.githubusercontent.com/13975226/214526570-d3cb8e2b-5e99-4641-8be1-b8d53358eb89.png">

- For sites whose backups are not in STOPPED state, it should work as existing
  <img width="734" alt="Screenshot 2023-01-25 at 2 58 23 PM" src="https://user-images.githubusercontent.com/13975226/214527099-fe676c7a-304c-433f-a0ae-cfd707cde90d.png">

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

rustamshaikh committed:
Hide backup scheduled or delayed if backups are stopped

- This hides the backup scheduled or delayed cards based on the storage level.
- If the usage level is set to Full, these cards will be hidden. Usage level defaults to Normal when not set.
- Uses :empty css selector to hide the card beacuse of the component hierarchy

rustamshaikh committed:
Fixes the progess bar when some backups were discarded

rustamshaikh committed:
Added unit tests for the backup scheduled and delayed show/hide cases

Initsogar commented on client/my-sites/backup/main.jsx:
```diff
@@ -202,11 +204,11 @@ function BackupStatus( {
 
 				<BackupDatePicker onDateChange={ onDateChange } selectedDate={ selectedDate } />
-				<BackupStorageSpace />
+				<BackupStorageSpace onUsageLevel={ setUsageLevel } />
```
@rustamshaikh I think maybe we can use a hook to manage the usageLevel state instead of passing it to multiple components. It might be useful for the next project as well.

What do you think?

rustamshaikh replied:
Thanks, @Initsogar for the suggestion. I will implement this using redux global state 👍

Initsogar reviewed (commented):
It looks great! Just left a suggestion. Let me know what do you think.

Initsogar reviewed (approved):
Approving the changes given that the suggestion is something we can do in a next iteration if preferred.

rustamshaikh committed:
Refactored to use global state for usageLevel

- Added the actions, reducers and selector to set and get usageLevel
- Fixed the test cases accordingly

rustamshaikh commented:
@Initsogar Done the changes for using the global state to store usage level.

rustamshaikh committed:
Fixed `StorageUsageLevelName` import

Initsogar commented on client/state/rewind/storage/reducer.ts:
```diff
@@ -0,0 +1,18 @@
+
+const initialState = {
+	usageLevel: StorageUsageLevels.Normal,
+	usage: null,
```
I think we can omit `usage` for now, as we are not considering it for this PR.

Initsogar commented on client/state/rewind/storage/types.ts:
```diff
@@ -0,0 +1,8 @@
+export type StorageUsageLevelName = 'Full' | 'Critical' | 'Warning' | 'Normal' | 'BackupsDiscarded';
```
Good idea to move it here :)

Initsogar reviewed (commented):
Overall, the code looks good! Thanks for addressing the concern about passing the usage level in multiple components.

Also, I think it would be great to add unit tests for the new reducer and selector for better maintainability. Let me know if you need any help with that 😊

rustamshaikh committed:
Removed unused variable `usage`

rustamshaikh committed:
Added test cases for the new selectors/reducers

rustamshaikh committed:
Corrected the outdated comment

rustamshaikh committed:
Updated comment

rustamshaikh committed:
Added additional test case

Initsogar reviewed (approved):
The last changes and tests looks good. Great work @rustamshaikh!

rustamshaikh committed:
Update message to correct singular/plural version

rustamshaikh committed:
Fixed failing test case
